AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -686,6 +686,18 @@ - (BOOL)isStateChangeValid:(ASVideoNodePlayerState)state
   return YES;
 }
 
+- (void)reset {
+  ASDN::MutexLocker l(__instanceLock__);
+  
+  if (_playerNode != nil) {
+    [_playerNode removeFromSupernode];
+    _playerNode = nil;
+  }
+  
+  [_player seekToTime:kCMTimeZero];
```

Comment:
@gazreese should we deallocate the AVPlayer as well?  This seems more consistent from a memory management perspective.